datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssignmentStrategy.java

```diff
@@ -44,28 +45,28 @@
   private static final int TASK_CAPACITY_UTILIZATION_PCT_DEFAULT = 90;
 
   private final PartitionThroughputProvider _throughputProvider;
-  private final DatastreamSourceClusterResolver _sourceClusterResolver;
   private final int _taskCapacityMBps;
   private final int _taskCapacityUtilizationPct;
   private final int _throughputInfoFetchTimeoutMs;
   private final int _throughputInfoFetchRetryPeriodMs;
 
+  // TODO Make this a config
+  private final boolean _enableThroughputBasedTaskCountEstimation = true;
+
 
   /**
    * Creates an instance of {@link LoadBasedPartitionAssignmentStrategy}
    */
   public LoadBasedPartitionAssignmentStrategy(PartitionThroughputProvider throughputProvider,
```

Comment:
One of the big missing piece for me is the metrics and alerts in this class. Please consider how you intend to debug the live setup when some issue happens. Some of the interesting pieces can be
a) How many times ThroughputProvider is invoked. This metric can go to Pinot specific Throughput Provider.
b) Current code will not generate alert for SREs to increase the maxTasks, as there is more demand based on throughput. `updateOrRegisterElasticTaskAssignmentMetrics` is currently tightly coupled with partition count, which will not work for throughput.

Reply:
Metrics/alerts will be published in a separate PR.